chapters/en/chapter10/3.mdx

```diff
@@ -0,0 +1,91 @@
+
+```python
+data = load_dataset("SetFit/ag_news", split="train")
+data.features()
+```
+
+These are the features of our dataset:
+
+```python out
+{'text': Value(dtype='string', id=None),
+ 'label': Value(dtype='int64', id=None),
+ 'label_text': Value(dtype='string', id=None)}
+```
+
+It contains a `text` and also some initial labels for the text classification. We'll add those to our dataset settings together with a `spans` question for the named entities:
+
+```python
+settings = rg.Settings(
+    fields=[rg.TextField(name="text")],
+    questions=[
+        rg.LabelQuestion(
+            name="label", title="Classify the text:", labels=data.unique("label_text")
+        ),
+        rg.SpanQuestion(
+            name="entities",
+            title="Highlight all the entities in the text:",
+            labels=["PERSON", "ORG", "LOC", "EVENT"],
+            field="text",
+        ),
+    ],
+)
+```
+
+Let's dive a bit deeper into what these settings mean. First, we've defined **fields**, these include the information that we'll be annotating. In this case, we only have one field and it comes in the form of a text, so we've choosen a `TextField`.
+
+Then, we define **questions** that represent the tasks that we want to perform on our data:
+
+- For the text classification task we've chosen a `LabelQuestion` and we used the unique values of the `label_text` column as our labels, to make sure that the question is compatible with the labels that already exist in the dataset.
+- For the token classification task, we'll need a `SpanQuestion`. We've defined a set of labels that we'll be using for that task, plus the field on which we'll be drawing the spans.
+
+To learn more about all the available types of fields and questions and other advanced settings, like metadata and vectors, go to the [Argilla docs](https://docs.argilla.io/latest/how_to_guides/dataset/#define-dataset-settings).
+
+## Upload the dataset
+
+Now that we've defined some settings, we can create the dataset:
+
+```python
+dataset = rg.Dataset(name="ag_news", settings=settings)
+
+dataset.create()
+```
+
+The dataset now appears in our Argilla instance, but you will see that it's empty. Now we need to add the records that we'll be annotating i.e., the rows in our dataset. To do that, we'll simply need to log the data as records and provide a mapping for those elements that don't have the same name in the hub and Argilla datasets:
+
+```python
+dataset.records.log(data, mapping={"label_text": "label"})
+```
+
+In our mapping, we've specified that the `label_text` column in the dataset should be mapped to the question with the name `label`. In this way, we'll use the existing labels in the dataset as pre-annotations so we can annotate faster.
+
+Now our dataset is ready to start annotating!
```

Comment:
I guess you've already considered the creation from UI. If you decided not to include it, I believe at least some kind of mention/explanation should be added, as it's the first thing shown to them, but there won't be anything about it in the guide they are following.

Reply:
I mention the creation from the UI at the very beginning of that page (before the "Configure your dataset" section), but maybe it's not clear enough? In any case, this specific dataset wouldn't work from the UI import because the labels are saved as strings rather than classLabels, so maybe it's better not to mention that here? 
Let me know what are your thoughts.

Reply:
I had missed that. Maybe to make it clearer, you can move that line to the end of the previous chapter where you say that you're gonna create the first dataset.